### ISSUE TITLE
[pcl] build failure

### ISSUE THREAD

**Issue by Bzz2745** [microsoft/vcpkg#38190]
### Operating system

Windows 11 

### Compiler

MSVC vC143, CUDA 12.4

### Steps to reproduce the behavior

```Shell
`./vcpkg install pcl[apps,core,cuda,opengl,openni2,qt,simulation,tools,visualization,vtk]:x64-windows`
```

### Failure logs

Installing 1/9 pcl[apps,core,cuda,opengl,openni2,qt,simulation,tools,visualization,vtk]:x64-windows@1.14.0#3...
Building pcl[apps,core,cuda,opengl,openni2,qt,simulation,tools,visualization,vtk]:x64-windows@1.14.0#3...
-- Using cached PointCloudLibrary-pcl-pcl-1.14.0.tar.gz.
-- Cleaning sources at C:/vcpkg/buildtrees/pcl/src/pcl-1.14.0-40680cf035.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/PointCloudLibrary-pcl-pcl-1.14.0.tar.gz
-- Applying patch add-gcc-version-check.patch
-- Applying patch fix-check-sse.patch
-- Applying patch fix-numeric-literals-flag.patch
-- Applying patch install-layout.patch
-- Applying patch install-examples.patch
-- Applying patch fix-clang-cl.patch
-- Applying patch gh-5926-exports.patch
-- Applying patch gh-5947-kinfu.patch
-- Applying patch gh-5985-inline.patch
-- Applying patch io_ply.patch
-- Using source at C:/vcpkg/buildtrees/pcl/src/pcl-1.14.0-40680cf035.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: C:/Users/bazunzhi/scoop/apps/cmake/3.29.2/bin/cmake.exe --build . --config Debug --target install -- -v -j13
    Working Directory: C:/vcpkg/buildtrees/pcl/x64-windows-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\pcl\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/pcl/portfile.cmake:80 (vcpkg_cmake_install)
  scripts/ports.cmake:175 (include)

### Additional context

[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/14972910/install-x64-windows-dbg-out.log)
this build failure may be caused by the version of CUDA still not supported by pcl.

**Comment by mvieth** [comment]
PCL 1.14.0 does not work with CUDA 12.4 because CUDA 12.4 was released two months after PCL 1.14.0.
On the PCL master branch, this is already fixed: https://github.com/PointCloudLibrary/pcl/commit/ccc4eee90a7242cf0d3c5e1ddb79716eabe4eb8c
You could try the vcpkg option `--head` to build the current PCL master branch instead, or replace CUDA 12.4 with CUDA 12.3

**Comment by Bzz2745** [comment]
> PCL 1.14.0 does not work with CUDA 12.4 because CUDA 12.4 was released two months after PCL 1.14.0. On the PCL master branch, this is already fixed: [PointCloudLibrary/pcl@ccc4eee](https://github.com/PointCloudLibrary/pcl/commit/ccc4eee90a7242cf0d3c5e1ddb79716eabe4eb8c) You could try the vcpkg option `--head` to build the current PCL master branch instead, or replace CUDA 12.4 with CUDA 12.3

OK,Thanks~
